### PR TITLE
Origin/feature/refund history

### PR DESCRIPTION
**Pull request by manic1990**
Added unit test cases for refund details

### CODE REVIEW

File: BookingSystem.Tests/ApiClient.Tests/Booking/APIClientGetBookingTest.cs [hunk 1]
```diff
@@ -49,7 +55,8 @@ public async void GetBooking_ShouldGetBookingRecord()
 
             mockFactory.Setup(a => a.CreateClient(It.IsAny<string>())).Returns(httpClient);
 
-            var apiClientRepository = new APIClientBookingRepository(mockFactory.Object);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/ApiClient.Tests/Booking/APIClientGetBookingTest.cs:52]: Keep URL in Appsettings

File: BookingSystem.Tests/ApiClient.Tests/Booking/APIClientSaveBookingTest.cs [hunk 2]
```diff
@@ -49,7 +55,8 @@ public async void SaveBooking_ShouldGetBookingRecord()
 
             mockFactory.Setup(a => a.CreateClient(It.IsAny<string>())).Returns(httpClient);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/ApiClient.Tests/Booking/APIClientSaveBookingTest.cs:56]: Keep url in Appsettings

File: BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = content,
+                });
+
+            var httpClient = new HttpClient(mockMessageHandler.Object);
+            httpClient.BaseAddress = new Uri("https://localhost:44348/api/");
+
+            mockFactory.Setup(a => a.CreateClient(It.IsAny<string>())).Returns(httpClient);
+
```
Comment (Backiaraj07) [inline BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs:56]: Keep url in appsetting

File: BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            var mockMessageHandler = new Mock<HttpMessageHandler>();
+            mockMessageHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.BadRequest,
+                    Content = null,
+                });
+
+            var httpClient = new HttpClient(mockMessageHandler.Object);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs:84]: Keep url in Appsettings

File: BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/CancelBookingTest.cs [hunk 5]
```diff
@@ -83,7 +83,7 @@ public async Task UpdateBookingAsync_ShouldUpdateBookingRecord()
                 context.Attributes.RemoveRange(context.Attributes);
                 context.SaveChanges();
 
-                refundRepository = new RefundRepository(context);
+                refundRepository = new DAL.Repository.RefundRepository(context);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/CancelBookingTest.cs:86]: Declare namespace in top.

**Review by Backiaraj07 — COMMENTED** [review]
Comments added

File: BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/CancelBookingTest.cs (hunk 5 above)
Comment (manic1990) [reply BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/CancelBookingTest.cs]: Resolved

File: BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs (hunk 4 above)
Comment (manic1990) [reply BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs:96]: Resolved

File: BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs (hunk 3 above)
Comment (manic1990) [reply BookingSystem.Tests/ApiClient.Tests/Refund/APIClientGetRefundTest.cs]: Resolved

File: BookingSystem.Tests/ApiClient.Tests/Booking/APIClientSaveBookingTest.cs (hunk 2 above)
Comment (manic1990) [reply BookingSystem.Tests/ApiClient.Tests/Booking/APIClientSaveBookingTest.cs:67]: Resolved

File: BookingSystem.Tests/ApiClient.Tests/Booking/APIClientGetBookingTest.cs (hunk 1 above)
Comment (manic1990) [reply BookingSystem.Tests/ApiClient.Tests/Booking/APIClientGetBookingTest.cs:52]: Resolved

**Review by Backiaraj07 — APPROVED** [review]
Code verified

**Comment by sathish199308** [comment]
Approved

**Pull request merged**